### ISSUE TITLE
Perform analysis on external link method and style

### ISSUE THREAD

**Issue by puzumaki** [UMNLibraries/design-system#13]
Some arguments on Code4Lib and Accessibility Slack for not using a symbol and explanatory text for external linking. Content in comments.

**Comment by puzumaki** [comment]
Code4Lib
Semiotics question: what does this icon [box with arrow pointing out upper right corner] typically mean to you if you saw it next to a link?
Evan 9:28 AM
That it opens in a new window (edit: or tab) (edited)
estelendur 9:29 AM
That it opens in a new window / that it opens a different site [3 agrees]
estelendur 9:29 AM
(As a user I generally hope that if I'm sent to a different site from somewhere that has a lot of site-internal links, like Wikipedia, that I get a new window/tab)
jrochkind  4 hours ago
does wikipedia do that, open new window/tab on external links?
estelendur  4 hours ago
I'm actually not sure? I'll check. (I habitually open all links in new tabs)
estelendur  4 hours ago
It does not. I am disappoint
jrochkind  4 hours ago
On desktop at least, I still am partial to “If I wanted it to open in a new window, I’d use command-click/right-click, by forcing it, you remove my option”
Which I think is still UX guidance from places like Nielsen.
But that may not match actual majority user expectations/wants these days. And may be especially so on mobile.
jrochkind  4 hours ago
that’s why I was curious if wikipedia had given in too, apparently not. :slightly_smiling_face: when wikipedia gives in, it’s time to completely throw in the towel on it haha. 
jrochkind  4 hours ago
i think it’s definitely a losing battle, but some are still waging it. https://medium.com/the-metric/links-should-open-in-the-same-window-447da3ae59ba
metageeky  4 hours ago
I really dislike the accessibility argument in that article. Yes, some SRs don't alert users. The problem is that there are solutions for providing context clues. Avoidance due to accessibility concerns is horrid.
metageeky  4 hours ago
We can't use images ever! No more buildings with stairs. Never write anything down. Don't say a word!
akorphan  4 hours ago
I think it's highly contextual whether a website ought to open things in new windows. I find myself caught by surprise more oftenw hen something DOESN'T open in a new window (usually after the fact, usually after I close the tab and realize I lost my source site's page)
akorphan  4 hours ago
I definitely have a mental model for when new-tab or new-window "should" happen but I have trouble articulating what that mental model might be
metageeky  4 hours ago
Exactly. It's very complicated and design heuristics are too broad a brush to make calls on this.
jrochkind  3 hours ago
What do you think the right way to make a call on it is?
akorphan  3 hours ago
User testing?
akorphan  3 hours ago
if it's "app-like" you could presumably have a setting, but even then you'd want to user-test the default state
hectorcorrea  3 hours ago
there are also security considerations when it comes to opening links in separate windows/tabs https://gosink.in/are-you-making-website-vulnerable-target-blank-or-performance/ (edited) 
:scream:
thatandromeda:computer:  3 hours ago
I’m team @jrochkind on this one, even if it means we’re sitting in our lawn chairs waving fists at the sky
:100:
jrochkind  2 hours ago
I would be happy to sit in lawn chairs and wave fists with a refreshing beverage with you and all others relevant sometime @thatandromeda (edited) 
jrochkind  2 hours ago
@hectorcorrea seems like a KIND of tenuous security consideration honestly, the author says “he has read some blogs and discussions” [no citations]  but “couldn’t do” reproducing  vulnerability cross-origin. (edited) 
thatandromeda:computer:  2 hours ago
:beers: :fist: :cloud:
hectorcorrea  1 hour ago
@jrochkind if the “Click this link to see the demo” link in the article did not convince you, then I certainly have nothing else to add.
Brian Holda 9:30 AM
Though I do know a librarian who was asking about what that icon meant when I felt it was self-explanatory. So a good lesson that even if we think it's fairly straightforward it is not to all.
metageeky 9:32 AM
This is exactly why I'm asking
akorphan 9:50 AM
It definitely is sometimes used for "external link" and sometimes used for "new window" and is always a bit confusing to me in that respect
jblyberg 11:27 AM
According to font awesome, it's an external link https://fontawesome.com/icons/external-link?style=duotone
Amy D 12:46 PM
Was out at meetings but ready to weigh in on the external vs. new tab. I use the arrow to indicate an external site, which I think is useful because it's a different structure (useful to know for keyboard users) and there aren't promises on that website that we made on ours (privacy, accessibility, security).
Regarding whether it opens in a new tab/window, I've differentiated that based on whether the website behaves like an app. Gmail, for instance, is a website-as-app. I expect those to open in a different tab automatically. I do not like having links open in new tabs by default because it takes control away from the user. Literally, it cannot be undone without some copy link to paste in address bar fun.
Amy D  35 minutes ago
And of course, seeing this in the accessibility slack as well brings to question everything.
akorphan  35 minutes ago
I'm not sure if I agree with "cannot be undone", though I guess I have a different interpretation of what it means to "undo" than you are going for
akorphan  34 minutes ago
i guess you mean, "user cannot override this design decision without extra labor as compared to the other design choice"
Amy D  34 minutes ago
How would you open a _blank link in the same tab?
akorphan  33 minutes ago
I mean, you're right about how I would open a _blank link in the same tab
akorphan  33 minutes ago
I just had a different gut interpretation of the "undo" usability test
akorphan  33 minutes ago
How do I "undo" opening a link in a new tab? ctrl-f4
akorphan  32 minutes ago
but your point is a valid one
Amy D  32 minutes ago
Undo wasn't quite the right word. :wink: More like "make a different choice" and "undo" the pre-programmed setting.
akorphan  32 minutes ago
right
akorphan  18 minutes ago
I guess I see the "greatest nuisance" test slgihtly differently: when something opens in the same tab when I want it to open in a new one, it's a greater nuisance to me than the opposite
akorphan  18 minutes ago
most of the time
metageeky  17 minutes ago
and that ties into the mixed semiotics of that icon.
akorphan  17 minutes ago
some of it has to do with ingrained behavior patterns to be sure. Startpage drives me bananas because it opens results in a new tab and discarded startpage tabs keep proliferating in my browser
akorphan  17 minutes ago
This comes back in part to the ideas about "app-like" vs "web-like" behavior, I think
metageeky  9 minutes ago
I miss the web that was about document sharing

**Comment by puzumaki** [comment]
Accessibility Slack
metageeky 9:28 AM
General semiotics question: what would you interpret this icon as meaning if you saw it next to a link?
geebru (He/Him)  4 hours ago
opens in a new tab/window
:heavy_plus_sign:
robinwhittleton  4 hours ago
Had this on GOV.UK to indicate that it was a link to an external site. We removed it as literally everyone in informal internal testing thought it meant ‘opens in a new tab’. (edited) 
:100: :+1:
metageeky  4 hours ago
That's exactly why I'm asking! A site uses it for external links and is asking how to convey the CSS generated icon to screen readers. Since it doesn't open a new tab, I think it's conveying false meaning.
robinwhittleton  4 hours ago
Found the PR: https://github.com/alphagov/govuk_frontend_toolkit/pull/293
This removes the styles for external links and their associated icons. It also removes the documentation. Will need discussion before a merge happens.
vavroom  4 hours ago
This idea of warning users that a link opens an external site is annoying me.
SO MANY clients confuse “opening in a new window” with “open an external site”.
I know for some countries (NZ in particular), warning a link goes to an external site is something that has to be done (for government sites). But… It’s not the same.
:+1:
robinwhittleton  4 hours ago
I wonder if NZ GOV does it because they lifted the GOV.UK codebase originally? (a good thing of course)
norah  4 hours ago
I'm also familiar with this icon as an "external site" indicator
metageeky  4 hours ago
Thank you Robin. I'm sharing that PR with the dev to explain the confusion.
:+1:
robinwhittleton  4 hours ago
Cool, once again ‘coding in the open’ proves worthwhile :slightly_smiling_face:
:1000:
vavroom  4 hours ago
@robinwhittleton Don’t think NZ Gov does it because of lifting codebase. At least not 5-6 years ago when I was auditing their site - it was an actual policy requirement
Charles  4 hours ago
the confusion is a little less for one of our clients where it means both external domain and external window.
metageeky  4 hours ago
is it always both?
Charles  3 hours ago
in the design system for one of our clients, yes
